Assets/Scripts/Towers/Tower.cs

```diff
@@ -1,152 +1,207 @@
+          
+          if (targetStats != null && Vector3.Distance(transform.position, collider.gameObject.transform.position) < Vector3.Distance(transform.position, target.gameObject.transform.position)) {
+            target = collider;
+          }
         }
-
-        VisualizeTargeting(target);
-
-        return target;
+      }
     }
-
-    private Enemy FindLowestPriorityEnemy() {
-        Enemy target = null;
-
-        foreach (Enemy enemy in enemiesInRange) {
-            if (target == null) {
-                target = enemy;
-            } else {
-                if (target.GetPriority() > enemy.GetPriority()) {
-                    target = enemy;
-                }
-            }
-        }
-
-        VisualizeTargeting(target);
-
-        return target;
+    
+    return target;
+  }
+  
+  private void PurgeDestroyedEnemies() {
+    for (int i = enemiesInRange.Count - 1; i >= 0; i--) {
+      if (enemiesInRange[i] == null) {
+        enemiesInRange.RemoveAt(i);
+      }
     }
-
-    private Enemy FindClosestEnemy() {
-        Enemy target = null;
-
-        foreach (Enemy enemy in enemiesInRange) {
-            if (target == null) {
-                target = enemy;
-            } else {
-                if (Vector3.Distance(transform.position, enemy.gameObject.transform.position) < Vector3.Distance(transform.position, target.gameObject.transform.position)) {
-                    target = enemy;
-                }
-            }
-        }
-
-        VisualizeTargeting(target);
-
-        return target;
+  }
+  
+  private void OnTriggerEnter(Collider other) {
+    if (other.CompareTag(TagConstants.ENEMY) && !enemiesInRange.Contains(other)) {
+      enemiesInRange.Add(other);
+      Debug.Log("Target");
     }
-
-    // This is purely for visualization purposes (i.e. serves no functional purpose)
-    private void VisualizeTargeting(Enemy target) {
-        if (currentTarget != target) {
-            if (currentTarget != null) {
-                if (currentTarget != null) {
-                    currentTarget.BeUntargeted();
-                }
-            }
-
-            if (target != null) {
-                target.BeTargeted();
-            }
-        }
+  }
+  
+  private void OnTriggerExit(Collider other) {
+    if (other.CompareTag(TagConstants.ENEMY) && enemiesInRange.Contains(other)) {
+      enemiesInRange.Remove(other);
     }
-
-    private void OnTriggerEnter(Collider other) {
-        if (other.TryGetComponent(out Enemy enemy)) {
-            enemiesInRange.Add(enemy);
-        }
-
-        SelectNewTarget();
+  }
+  
+  private void OnEnemyHit(GameObject enemyObject) {
+    if (towerType.areaEffectType != null) {
+      GameObject areaEffectObject = Instantiate(towerType.areaEffectType.areaEffectPrefab, enemyObject.transform.position, Quaternion.identity);
+      areaEffectObject.TryGetComponent(out AreaEffect areaEffect);
+      areaEffect.SetAreaEffectType(towerType.areaEffectType);
```

Comment:
Couldn't the areaEffectPrefab have the areaEffectType already pre-set, since it is reached through the areaEffectType anyways?

Reply:
Fair enough, I thought it would be neater to have everything tied to the type, but it's a little awkward